### PR TITLE
[docs] Files and formats

### PR DESCRIPTION
**Pull request by stevhliu**
This PR attempts to clarify the difference between file types (`safetensors`, `ckpt`, `bin`) and storage layout (multifolder,  single file) to avoid confusion when discussing "Diffusers" format and "single file" format. The most common misconception is that `safetensors` is just a file type, and single file or multifolder layout can both hold `safetensors` files. The PR also removes the old "Load safetensors" doc and discusses it in the current one here where it is more contextually relevant.

todo:
~- [ ] add Keras section once Space is fixed~

### CODE REVIEW

File: docs/source/en/using-diffusers/other-formats.md [hunk 1]
```diff
@@ -10,167 +10,218 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
 specific language governing permissions and limitations under the License.
 -->
 
-# Load different Stable Diffusion formats
+# Model files and formats
 
 [[open-in-colab]]
 
-Stable Diffusion models are available in different formats depending on the framework they're trained and saved with, and where you download them from. Converting these formats for use in 🤗 Diffusers allows you to use all the features supported by the library, such as [using different schedulers](schedulers) for inference, [building your custom pipeline](write_own_pipeline), and a variety of techniques and methods for [optimizing inference speed](../optimization/opt_overview).
+Diffusion models are saved in various file types and organized in different layouts. Diffusers prefers storing model weights as `.safetensors` files in a multifolder layout, but it also supports loading file types (can be `.safetensors`, `.ckpt`, etc.) from a single file layout which is commonly used in the diffusion ecosystem.
```
Comment (yiyixuxu) [inline docs/source/en/using-diffusers/other-formats.md:17]: should we define the multifolder layout "diffusers-style format" and the other one "single-file format" and use these terms in the post? These are the terms already used by the community 

While we are trying to support loading the single-file format as first-class citizen, the multifolder layout is our official format, we prefer this format, and all our models are released in this format

File: docs/source/en/using-diffusers/other-formats.md [hunk 2]
```diff
@@ -10,167 +10,218 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
+    use_safetensors=True,
+).to("cuda")
 ```
 
-Then, you can generate an image like:
+Benefits of using the multifolder layout include:
 
-```py
-from diffusers import DiffusionPipeline
+1. Faster to load each component file individually or in parallel.
```
Comment (yiyixuxu) [inline docs/source/en/using-diffusers/other-formats.md:155]: diffusers-style format provides much greater flexibility and allows users to customize loading. We should expand on this section and support it with more practical and convincing examples! (the example here demonstrates the API they can use to customize loading, but it is very simple and it does not directly demonstrate the speed and memory-saving advantage, users have to deduce that themselves)

if you want to use two different models in single_file, most likely they will share some components, e.g. vae, text_encoders etc, and you will need to load these models twice - it's slower and they take up more memories than needed.  With diffusers-format, you would load one model, and load only the different components for the second one, and use the `from_pipe` API to switch.  that  cc @asomoza here. Can you help think of an example that will be meaningful for the single-format fans? 

another strong argument we can make is when the model includes large components - I wonder how it is even possible for them to handle models like sd3 in a single-file format? with 3 text encoders (including t5) and a 8b unet? also cc @asomoza here for insights -

**Review by yiyixuxu — COMMENTED** [review]
I left some comments
thanks a ton for this PR! it is a very important doc to us:)

File: docs/source/en/using-diffusers/other-formats.md (hunk 2 above)
Comment (asomoza) [reply docs/source/en/using-diffusers/other-formats.md:155]: It's mostly the same that you mentioned, here are the "maybe" reasons:

- For SDXL we can literally use just one [vae](https://huggingface.co/madebyollin/sdxl-vae-fp16-fix) for all the models, so if you have 10 models, you will save 3.5 GB of space and bandwidth.
- For SD 1.5 there's like 4 or 5 good vae models, they're smaller but even with that, since we can switch them on the fly, we can always use the optimal one for each model.

So all the models could be distributed without vaes, in sites like civitai.com there's always a confusion with models that come with or without vae, this is not a problem with the diffusers format.

For the SDXL and SD 1.5 text encoders, we could save that space, but we really don't know which models have trained the text encoders or not, so this is harder to make evident to the single-file fans unless the model owners and sharing sites start to make that information visible.

These ones are the "strong" reasons:

- Turbo, Lighting and Hyper-SD models, these ones just change the unet, so there's no reason to share anything else.
- New models with T5 (Pixart Sigma, Kandinsky 3, SD3), the text encoder alone has a size between 17GB and 22 GB, It’s simply not feasible to share the text encoder with every fine-tuned model, and probably no one will train them either. There’s really no need to do so.

IMO the last one is the main reason people will be forced to stop sharing single-file checkpoints.

File: docs/source/en/using-diffusers/other-formats.md (hunk 1 above)
Comment (stevhliu) [reply docs/source/en/using-diffusers/other-formats.md]: I think it may be better to reframe it as a "multifolder" and "single-file" layout instead of the current terms used by the community because:

1. It more accurately describes how the files are stored, avoiding confusion when discussing files and formats. For example, we talk about files being stored in the safetensors format, but when we talk about the "Diffusers-style" format, it may lead people to think we are talking about a file type rather than just the layout of how the files are organized.
2. It is more neutral because it doesn't position it as Diffusers vs single-file (community).

Comment (yiyixuxu) [reply docs/source/en/using-diffusers/other-formats.md]: Can we call it "diffuser-style multifolder"? I found it confusing without "diffusers" in the term:) 

It is a simple and plain fact that it is our format, and there is always a very clear difference from single-file. We will support single-file as best as we can, but the single-file checkpoints will always come from the community. And all the models released by us will be in this multifolder format.

Comment (stevhliu) [reply docs/source/en/using-diffusers/other-formats.md]: Updated to "Diffusers-multifolder" layout everywhere :)

**Comment by stevhliu** [comment]
We can add the Keras section back whenever @sayakpaul gets a chance to fix it (no rush!), and I think it'd also be helpful to move the content from [here](https://huggingface.co/docs/diffusers/main/en/api/loaders/single_file#loading-pipelines-and-models-via-fromsinglefile) (usage examples for `from_single_file` in the API docs) to this doc in a separate PR. WDYT?

**Comment by sayakpaul** [comment]
> We can add the Keras section back whenever @sayakpaul gets a chance to fix it (no rush!),

Yeah sure. Actually, the problem is with the installation and the code itself.

**Review by yiyixuxu — APPROVED** [review]
good to merge now!
want to follow up a little bit more on recommendations on how to use diffusers checkpoint in order to maximize the efficiency. I think we need to refactor the from_single_file a little bit for that so will be a future PR :)
